src/lava/magma/core/model/py/ports.py

```diff
@@ -282,22 +525,32 @@ def __init__(self,
 
     @property
     def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
+        """Property to get all csp ports used by the port.
+
+        Returns
+        -------
+        A list of all csp ports used by the port.
+        """
         if self._csp_send_port is not None and self._csp_recv_port is not None:
             return [self._csp_send_port, self._csp_recv_port]
         else:
             # In this case the port was not connected
             return []
 
+    @abstractmethod
     def service(self):
+        """Abstract method to set the received value to the given Var or sends
```

Comment:
Incomplete explanation. What received value? For this to make sense, you have to give a refresher what a VarPort is and how it work. Not super long but briefly.

Reply:
Added explanation.